keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/VerificationRepositoryTests.kt

```diff
@@ -377,4 +382,101 @@ abstract class VerificationRepositoryTests<IMPLEMENTATION : VerificationReposito
       next().isSuccess().hasSize(1)
     }
   }
+
+  //
+  // getStatesBatch tests
+  //
+  @Test
+  fun `no verification states`() {
+    context.setup()
+    val contexts = listOf(context)
+
+    expectThat(subject.getStatesBatch(contexts))
+      .hasSize(contexts.size)
+      .containsExactly(emptyMap())
+  }
+
+  @Test
+  fun `one verification state`() {
+    context.setup()
+    val contexts = listOf(context)
+
+    subject.updateState(context, verification, PENDING)
+
+    val stateMaps = subject.getStatesBatch(contexts)
+
+    expectThat(stateMaps)
+      .hasSize(contexts.size)
+
+    val stateMap = stateMaps.first()
+    expectThat(stateMap)
+      .containsKey(verification.id)
+
+    expectThat(stateMap[verification.id])
+      .isNotNull()
+      .with(VerificationState::status) { isEqualTo(PENDING) }
+  }
+
+  @Test
+  fun `multiple contexts, multiple verifications`() {
+    //
+    // setup
+    //
+    val c1 = context
+    val c2 = VerificationContext(
+      deliveryConfig = deliveryConfig,
+      environmentName = "test",
+      artifactReference = "fnord-docker-stable",
+      version = "fnord-0.191.0-h379.fbde246"
+    )
+
+    val v1 = verification
+    val v2 = DummyVerification("2")
+
+    val contexts = listOf(c1, c2)
+    contexts.forEach { it.setup() }
+
+    // First context: v1: PASS, v2: FAIL
+    subject.updateState(c1, v1, PASS)
+    subject.updateState(c1, v2, FAIL)
+
+    // Second context: v1: PENDING, v2: PENDING
+    subject.updateState(c2, v1, PENDING)
+    subject.updateState(c2, v2, PENDING)
+
+
+    //
+    // Code under test
+    //
+    val result = subject.getStatesBatch(contexts)
+
+    expect {
+      that(result)
+        .hasSize(contexts.size)
+
+      that(result[0])
+        .and {
+          get { get(v1.id) }
+          .isNotNull()
+          .with(VerificationState::status) { isEqualTo(PASS) }
+        }
+        .and {
+          get { get(v2.id) }
+          .isNotNull()
+          .with(VerificationState::status) { isEqualTo(FAIL) }
+        }
+
+      that(result[1])
+        .and {
+          get { get(v1.id) }
+            .isNotNull()
+            .with(VerificationState::status) { isEqualTo(PENDING) }
+        }
+        .and {
+          get { get(v2.id) }
+            .isNotNull()
+            .with(VerificationState::status) { isEqualTo(PENDING) }
```

Comment:
Or even simpler: 😊
```suggestion
            .get { status }.isEqualTo(PENDING)
```

Reply:
That's actually considerably slower because it does some code reading tomfuckery to generate the string representation of the path (that will get used if the assertion fails). When using a method reference it can just use its name. Unfortunately there's not really a way round that as you can't use reflection to parse the content of the lambda. You can do `get("status") { status }` which will avoid the code reading trick, but that's kinda ugly.